cornell-dti/course-plan
# DOM manipulation removal

### Summary 

Remove most of the existing bad DOM Manipulation code in the codebase and replace it with event handling per issue #10. This will make it much easier in the future to see and change how data flows between components.

- [x] Removed almost all instances of `document.getElementByID` or `.innerHTML` that grabs and manipulates data poorly, especially in opening/closing modals
- [x] Removed almost all `$parent` references.
- [x] Removed other outdated/duplicated code and comments (2 `<semester>`s, NewCustomCourse.vue, etc.)
- [ ] Add code that allows you to click outside of modals to close them without DOM Manipulation ([see Monday](https://courseplan.monday.com/boards/433956786/pulses/848083188))
- [ ] Refactor the add course modal to avoid using DOM Manipulation in `Modal.vue`, `NewCourse.vue`, and `OnboardingTransfer.vue` (waiting for @hahnbeelee's work to be finished before messing around with this file, will take significant more work than these other elements, [see Monday](https://courseplan.monday.com/boards/433956786/pulses/848072269))
- [ ] Refactor the direction of the triple dot menu on semesters (still need a better solution for this, [see Monday](https://courseplan.monday.com/boards/433956786/pulses/542946252))
- [ ] Refactor `createCourse` so it is not called with `$parent` (@SamChou19815 do you want to take this?)

### Test Plan 

Ensure that everything works on the site compared to master (https://cornelldti-courseplan-dev.web.app/) with the exception of clicking outside of modals closing them.

Check if all of the following modals can be opened, closed, and used as expected
- Add Course
- New Semester
- Edit Semester
- Delete Semester

### Notes 

The rest of the DOM Manipulation should be removed so it is not duplicated in the future like it has been since CoursePlan started. 

### Breaking Changes  

As specified above, just clicking outside of modals.
